x/accesscontrol/keeper/keeper.go

```diff
@@ -206,13 +207,40 @@ func (k Keeper) GetWasmDependencyAccessOps(ctx sdk.Context, contractAddress sdk.
 	return selectedAccessOps.ToSlice(), nil
 }
 
+func ParseContractReferenceAddress(maybeContractAddress string, sender string, msgInfo *types.WasmMessageInfo) string {
+	// sender in case the use case is expected to be one contract calling another expecting a separate call
+	const reservedSender = "_sender"
+	if maybeContractAddress == reservedSender {
+		return sender
+	}
+	// parse the jq instruction from the template - if we can't then assume that its ACTUALLY an address
+	op, err := jq.Parse(maybeContractAddress)
+	if err != nil {
+		return maybeContractAddress
+	}
+	// retrieve the appropriate item from the original msg
+	data, err := op.Apply(msgInfo.MessageFullBody)
+	// if we do have a jq selector but it doesn't apply properly, return maybeContractAddress
+	if err != nil {
+		return maybeContractAddress
+	}
+	// if we parse it properly convert to string and return
+	var newValBytes string
+	err = json.Unmarshal(data, &newValBytes)
+	if err != nil {
+		return maybeContractAddress
+	}
+	return newValBytes
```

Comment:
if this returns a nested json, the `AccAddressFromBech32` below would error. In that case do we want to propagate the error or use maybeContractAddress instead?

Reply:
I think it would be correct to error, since that would indicate that the wrong JQ selector path was used and while it was a valid path it didn't build a proper contract address